### ISSUE TITLE
Addressing first Monterey evaluation issues

### ISSUE THREAD

**Issue by adriatic** [monterey-framework/monterey#286]
This article is the summary of all Rob's feedback followed by the action items needed to fully address the issues brought up in that feedback. As such this article should stay open until all problems are addressed.

# Comments addressing existing version

### Adding style
> Regarding the “new project” experience, this has come a long way. The only nitpicks I would have are minor UI issues. Maybe some style improvements, etc.

Rather than having a single, platform independent look and feel, we should consider supporting different LAF on Mac, different LAF on Windows and different LAF on Linux - if we come to the conclusion that this is within our ability to do and if we like this approach. I would further encourage a discussion on the meaningfulness of having user settable styles as well.

Let's be ready to accommodate all people's opinions on styling as we are switching from technical correctness to a lot more subjective field of aesthetics - a lot more difficult problem to solve for us.

___This discussion continues [here](https://github.com/monterey-framework/monterey/issues/285)___
***

### Expanding current functionality (scaffolding)
> The biggest area that seems to need some improvement is what happens after the application is created. For example, I stopped the tasks, and then when I wanted to use the app again, I wasn’t quite sure what to do. I figured out I needed to start up two tasks again to get it working. Not a big deal but I wonder if some improvement can be made there. Perhaps some standard grouping of the two tasks and a button to launches them both? I’m not sure.

___This discussion continues [here](https://github.com/monterey-framework/monterey/issues/290)___.

***

### Live help
> Maybe think about building some automated help in so that when someone creates their first project, the help system shows them the different features of Monterey and explains how they can be used during development. I found myself a bit overwhelmed by all the plugins and wasn’t sure where to start.

I think that this type of the guidance should be a composition of two independent approaches:

- application being aware of the user context should always strive to show only parts of UI that is relevant to the context. We already have the beginning of this approach implemented in the main view (show irrelevant tiles check box) and I do not believe that this is all we can do

- application being aware of the user context should behave as a text editor with intellisense.

Both types of assistance should be enabled / disabled as a user preference.

___This discussion continues [here](https://github.com/monterey-framework/monterey/issues/282)___

**Comment by adriatic** [comment]
# Comments addressing future versions

### Development services
> I’d like to hear more about the development services ideas you have

### Loaders
> How is the module loader is related development services. If possible, it would be best to come up with a solution that wasn’t tied to a particular loader. The CLI uses RequireJS as its default today, however, it has been designed so that in the future it can support other loaders such as SystemJS (without jspm or system-builder) as well as the WHATWG Reflect.Loader.

**Comment by JeroenVinke** [comment]
> Expanding current functionality (scaffolding)
> The biggest area that seems to need some improvement is what happens after the application is created. For example, I stopped the tasks, and then when I wanted to use the app again, I wasn’t quite sure what to do. I figured out I needed to start up two tasks again to get it working. Not a big deal but I wonder if some improvement can be made there. Perhaps some standard grouping of the two tasks and a button to launches them both? I’m not sure.

Lets make sure that we are looking at the same problem to solve.

Assuming that we are wanting to launch an application we scaffolded earlier (we have closed monterey or restarted our PC). You would have to do the following:

1. launch monterey
2. click the TaskManager button in the toolbar
3. Select the project I want to launch
4. Click on `gulp watch`  (the taskmanager switches to the task detail view, showing the progress of the gulp watch process)
5. Select the project again
6. Click on `dotnet run` 
7. Close the taskmanager
8. Start up the browser (via the Chrome tile for example)

Steps 2 through 7 can done much more quickly if we add a "Run" tile to Monterey, which starts the `gulp watch` and `dotnet run` tasks in the background. Meaning that you see `2 running processes` in the taskbar, but you do not go to the taskmanager popup.

**Comment by adriatic** [comment]
It will never stop amazing me how we think about the same issues at the (nearly) same time - check https://github.com/monterey-framework/monterey/issues/290 as an example

**Comment by JeroenVinke** [comment]
The reason why I think we should synchronize about what the actual problem is is this:

https://github.com/monterey-framework/monterey/issues/289 explains that we should extend monterey with the feature to install plugins (with ai-plugin). I do not see how this reduces the amount of steps involved to take an **existing** application and start it. By "starting" I mean to run `gulp watch` and `dotnet run` at the same time. Opening editors and browsers is not part of this.

https://github.com/monterey-framework/monterey/issues/290 shows the currently existing process of running a **new** application. This again does not reduce the amount of steps to get the project running.

So are we seeing the same problem here?

**Comment by JeroenVinke** [comment]
Don't get me wrong, I completely agree with what's in https://github.com/monterey-framework/monterey/issues/289, but I do not see how this relates to the feedback we received:

> The biggest area that seems to need some improvement is what happens after the application is created. For example, I stopped the tasks, and then when I wanted to use the app again, I wasn’t quite sure what to do. I figured out I needed to start up two tasks again to get it working. Not a big deal but I wonder if some improvement can be made there. Perhaps some standard grouping of the two tasks and a button to launches them both? I’m not sure.

**Comment by adriatic** [comment]
I know that we are in a complete agreement - it's only that you do not know that yet, as I need to provide some additional explanations **coming immediately** below:

**Comment by adriatic** [comment]
I am indeed expanding on this starting paragraph, that I wrote at the beginning of this discussion.

![image](https://cloud.githubusercontent.com/assets/2712405/18395814/5a4796bc-768e-11e6-859b-f3756979ea8c.png)

However not being sure what can we agree on to be in the "first release" responding to Rob's comments and what will go in the "second (subsequent) release" I disregarded this timing problem completely and wanted to show the trajectory towards the point where Monterey can handle a previously generated application using the [Aurelia-CLI based generator](https://aurelia-ui-toolkits.gitbooks.io/monterey-user-manual/content/creating_new_application/aurelia-cli.html).


In order to allow for the two releases approach (all addressing current Rob's comments, I defined these two releases as:

1. [Extending Monterey beyond being "app generator" - proposal 2](https://github.com/monterey-framework/monterey/issues/290) to be in the **first** release. This functionality supports adding more code to the existing app - without being able to add mode third party modules.

2. [Extending Monterey beyond being "app generator" - proposal 1](https://github.com/monterey-framework/monterey/issues/289) to be in the **second** release). This functionality adds the support for adding third party modules as well.

My own preference would be merge both releases into one, which addresses Rob's concern stated as

> The biggest area that seems to need some improvement is what happens after the application is created.

by adding more than he asked for - nearly full support for subsequent evolution of the app created with Aurelia-CLI generator (using just RequireJS module loader.

Does this explain my intent?
